WordPress/WordPressTest/StoreCoordinatorTests.swift

```diff
@@ -56,32 +62,74 @@ class StoreCoordinatorTests: XCTestCase {
+        let payment: PendingPayment = (premium.id, testProduct, testSite)
+        let coordinator = storeCoordinator(paymentsEnabled: true, pending: payment)
+        let product = productForPlan(TestPlans.business.plan)
+
+        // And now attempt a second purchase
+        XCTAssertThrowsError(try coordinator.purchasePlan(business, product: product, forSite: testSite))
+    }
+    
+    func testCanMakePaymentWhenNoPaymentIsPending() {
+        let coordinator = storeCoordinator(paymentsEnabled: true, pending: nil)
+        let product = productForPlan(TestPlans.business.plan)
+
+        do {
+            try coordinator.purchasePlan(business, product: product, forSite: otherSite)
+        } catch {
+            XCTFail("Expected call not to throw")
+        }
+    }
+    
     // ========================= END OF TESTS =============================== //
 
 
     // MARK: - Helpers
 
     private func availabilityWith(plan plan: Plan, active: Plan, paymentsEnabled: Bool, pendingState: PendingState) -> PurchaseAvailability {
-        let pendingPayment = pending(plan: plan, siteID: testSite, state: pendingState)
+        let pendingPayment = pending(plan: plan, productID: testProduct, siteID: testSite, state: pendingState)
         let coordinator = storeCoordinator(paymentsEnabled: paymentsEnabled, pending: pendingPayment)
         return coordinator.purchaseAvailability(forPlan: plan, siteID: testSite, activePlan: active)
     }
 
-    private func storeCoordinator(paymentsEnabled paymentsEnabled: Bool, pending: (Plan, Int)?) -> StoreCoordinator<MockStore> {
+    private func storeCoordinator(paymentsEnabled paymentsEnabled: Bool, pending: PendingPayment?) -> StoreCoordinator<MockStore> {
         var store = MockStore.succeeding()
         store.canMakePayments = paymentsEnabled
+ 
         let coordinator = StoreCoordinator(store: store)
-        coordinator.pendingPayment = pending
+        
+        if let pending = pending,
+            let plan = TestPlans.allPlans.filter({ $0.id == pending.planID }).first {
+            let product = productForPlan(plan)
+            try! coordinator.purchasePlan(plan, product: product, forSite: pending.siteID)
+        }
+        
         return coordinator
     }
+    
+    private func clearUserDefaults() {
+        let defaults = NSUserDefaults.standardUserDefaults()
+        defaults.removeObjectForKey("PendingPaymentPlanIDUserDefaultsKey")
+        defaults.removeObjectForKey("PendingPaymentProductIDUserDefaultsKey")
+        defaults.removeObjectForKey("PendingPaymentSiteIDUserDefaultsKey")
+    }
+    
+    private func productForPlan(plan: Plan) -> MockProduct {
+        return MockProduct(localizedDescription: plan.tagline,
+                           localizedTitle: plan.title,
+                           price: 299.99,
+                           priceLocale: NSLocale.currentLocale(),
+                           productIdentifier: plan.productIdentifier!)
+    }
```

Comment:
What if we move this to `TestPlans.swift`? Also, I don't think we're testing price formatting here, but I'd be more explicit than `currentLocale()`. Otherwise [subtle bugs](https://github.com/wordpress-mobile/WordPress-iOS/pull/5280#issuecomment-216820029) might happen 😉